src/Chapter3.hs

```diff
@@ -373,6 +379,26 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data Fighter = Fighter {
+  fighterHealth :: Int
+  , fighterAttack :: Int
+  , fighterGold :: Int
+}
+
+attackFatal :: Fighter -> Fighter -> Bool
+attackFatal atk def
+    | health - attack <= 0 = True 
+    | otherwise = False
+    where 
+      health = fighterHealth def
+      attack = fighterAttack atk 
+
+fight :: Fighter -> Fighter -> Int
```

Comment:
Nice fight! ⚔️ 
The only issue with reusages of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two type for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Fighter
newtype Knight = Knight Fighter
```